File: src/main/java/nikol/product/ProductView.java
package nikol.product;

public class ProductView extends AbstractProduct {
    public ProductView(int id, String name, double price) {
        super(id, name, price);
    }

    public String toString() {
        return "id: " + id + ", name: " + name + ", price: " + price;
    }
}

File: src/main/kotlin/ProductView.kt
package voidpointer.mc.tests

data class ProductView(
    override var id: Int? = null,
    override var name: String,
    override var description: String,
    var price: Double
) : AbstractProduct() {

    override fun equals(other: Any?): Boolean {
        return super.equals(other)
    }

    override fun hashCode(): Int {
        return AbstractProduct.hashCode()
    }
}
